python/ray/data/BUILD

```diff
@@ -19,3 +19,15 @@ py_test_module_list(
   tags = ["team:core", "exclusive"],
   deps = ["//:ray_lib"],
 )
+
+py_test_module_list(
+  files = [
+    "tests/test_sort.py",
+  ],
+  size = "small",
+  extra_srcs = SRCS,
+  name_suffix = "_push_based",
```

Comment:
Can you make the shuffle impl a DatasetContext option following the fusion optimization options? That will make it easier to write as a parameterized test without needing bazel env var hacks.